### PR TITLE
Performance test automation for single node

### PR DESCRIPTION
**Pull request by inthirakumaaran**
This PR contains scripts to deploy a single node environment, setup IS server in IS node and create tables in the RDS

### CODE REVIEW

File: cloudformation/start_performance_single.sh [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+wso2_is_1_ip="$(aws ec2 describe-instances --instance-ids $wso2is1_instance | jq -r '.Reservations[].Instances[].PrivateIpAddress')"
+echo "WSO2 IS Node Private IP: $wso2_is_1_ip"
+
+echo ""
+echo "Getting RDS Hostname..."
+rds_instance="$(aws cloudformation describe-stack-resources --stack-name $stack_id --logical-resource-id WSO2ISDBInstance2 | jq -r '.StackResources[].PhysicalResourceId')"
+rds_host="$(aws rds describe-db-instances --db-instance-identifier $rds_instance | jq -r '.DBInstances[].Endpoint.Address')"
+echo "RDS Hostname: $rds_host"
+
+copy_isserver_edit_command="scp -i $key_file -o "StrictHostKeyChecking=no" $home/isserver_edit.sh ubuntu@$bastion_node_ip:/home/ubuntu/"
```
Comment (vihanga-liyanage) [inline cloudformation/start_performance_single.sh]: We don't need to copy $home/setup, isserver_edit.sh, setup_is.sh and createDB.sql files explicitly from here. We're coping the distribution anyway. Please change the logic to take the files from there.

Comment (vihanga-liyanage) [reply cloudformation/start_performance_single.sh]: Also, let's remove the unnecessary files from this branch and use it for single node performance testing.

File: cloudformation/start_performance_single.sh [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        ParameterKey=WSO2InstanceType,ParameterValue=$wso2_is_instance_type \
+        ParameterKey=BastionInstanceType,ParameterValue=$bastion_instance_type \
+    --capabilities CAPABILITY_IAM"
+
+echo ""
+echo "Creating stack..."
+echo "$create_stack_command"
+stack_id="$($create_stack_command)"
+stack_id=$(echo $stack_id|jq -r .StackId)
+# stack_id="arn:aws:cloudformation:us-east-1:610968236798:stack/is-performance-test-stack-single-node/786da520-359d-11e9-99be-0e94a1882570"
```
[the remark this answers is not in the document]
Comment (inthirakumaaran) [reply cloudformation/start_performance_single.sh]: resolved

File: cloudformation/start_performance_single.sh [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+
+
+# Get absolute paths
+key_file=$(grealpath $key_file) 
+
+echo "your key is"
+echo $key_file
+
+
```
[the remark this answers is not in the document]
Comment (inthirakumaaran) [reply cloudformation/start_performance_single.sh]: resolved

File: cloudformation/start_performance_single.sh [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+home="$results_dir/setup/single-node-setup"
+
+echo $home
+
+# estimate_command="$results_dir/jmeter/run-performance-tests.sh -t ${run_performance_tests_options[@]}"
+# echo ""
+# echo "Estimating time for performance tests: $estimate_command"
+# # Estimating this script will also validate the options. It's important to validate options before creating the stack.
+# $estimate_command
```
[the remark this answers is not in the document]
Comment (inthirakumaaran) [reply cloudformation/start_performance_single.sh]: resolved

**Pull request merged**